client/src/pages/SignUp/SignUp.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useState } from 'react'
+
+// styled-components
+import {
+  MainContainer,
+  WelcomeText,
+  InputContainer,
+  ButtonContainer,
+  Login
+} from "./SignUpStyles";
+
+// components
+import { SignUpInput } from '../../components/Input/InputStyles';
+import { Button } from '../../components';
+import { Navigate } from 'react-router-dom';
+
+function SignUp() {
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+  const [email, setEmail] = useState('');
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [redirect, setRedirect] = useState(false)
+
+  const submit = async (e: Event) => {
+    e.preventDefault();
+    await fetch("/api/user/register", {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({
+        username,
+        email,
+        password,
+        firstName,
+        lastName
```

Comment:
These should be snake case, so `first_name` and `last_name`. You can look these up in the server calls api in the wiki.

Reply:
I just realized that this was my bad, sorry haha. Go ahead and change it, I will make sure the wiki is updated. thanks!